crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -113,19 +115,43 @@ impl<'a> ObjectRuntime<'a> {
         }
     }
 
-    pub fn new_id(&mut self, id: ObjectID) {
+    pub fn new_id(&mut self, id: ObjectID) -> PartialVMResult<()> {
+        if self.state.new_ids.len() == (MAX_NUM_NEW_IDS as usize) {
```

Comment:
Instead of casting to usize, would it be easier to make the constant in sui-protocol-constants the type you want here (`usize`) in the first place?

Reply:
done